src/pages/AddLiquidity/WrappedCurrencyToggle.tsx

```diff
@@ -0,0 +1,65 @@
+import { Trans } from '@lingui/macro'
+import { useWeb3React } from '@web3-react/core'
+import { ToggleElement, ToggleWrapper } from 'components/Toggle/MultiToggle'
+import { useCurrency } from 'hooks/Tokens'
+
+import { WRAPPED_NATIVE_CURRENCY } from '../../constants/tokens'
+
+function isNative(currencyId?: string) {
+  return currencyId === 'ETH'
+}
+
+function isWrappedNative(currencyId?: string, chainId?: number) {
+  return chainId !== undefined && currencyId === WRAPPED_NATIVE_CURRENCY[chainId]?.address
+}
+
+type WrappedCurrencyToggleProps = {
+  currencyIdA?: string
+  currencyIdB?: string
+  onChangeCurrencies: (CurrencyAId?: string, currencyBId?: string) => void
+}
+
+export function WrappedCurrencyToggle({ currencyIdA, currencyIdB, onChangeCurrencies }: WrappedCurrencyToggleProps) {
+  const { chainId } = useWeb3React()
+
+  // Get native and wrapped native Currencies
+  const native = useCurrency('ETH', chainId)
+  const wrappedNative = chainId ? WRAPPED_NATIVE_CURRENCY[chainId] : undefined
+
+  // checking to see if currencyIdA of currencyIdB is native or wrapped native
+  const currencyIsNative = isNative(currencyIdA) || isNative(currencyIdB)
+  const currencyIsWrappedNative = isWrappedNative(currencyIdA, chainId) || isWrappedNative(currencyIdB, chainId)
+
+  if (!currencyIsNative && !currencyIsWrappedNative) return null
+
+  const handleToggle = () => {
+    if (currencyIsNative) {
+      if (isNative(currencyIdA)) onChangeCurrencies(wrappedNative?.address, currencyIdB)
+      else onChangeCurrencies(currencyIdA, wrappedNative?.address)
+    }
+
+    if (currencyIsWrappedNative) {
+      if (isWrappedNative(currencyIdA)) onChangeCurrencies('ETH', currencyIdA)
+      else onChangeCurrencies(currencyIdA, 'ETH')
+    }
+  }
+
+  if (!currencyIsNative && !currencyIsWrappedNative) return null
+
+  return (
+    <>
+      {native && wrappedNative ? (
```

Comment:
double check that the native and wrappedNative checks here are necessary--it looks like the line above may already be handling this case? (Though maybe that should be `||` instead of `&&`?)

```suggestion
  if (!native || !wrappedNative) return null
  return (
```

this lets you substantially simplify the template

Reply:
Hey @JFrankfurt,

isCurrencyNative & isCurrencyWrappedNative are just checks to make sure one of the currencies provided (currencyIdA or currencyIdB) are native or wrapped.

native and wrappedNative are the actual wrapped currencies to shown in the toggle the conditional is to make sure they are loaded in properly

I made is more clear in my most recent commit